Spike/run script

connoratrug opened the pull request:
Initial setup to run script as package based on https://gist.github.com/SietsmaRJ/fd4ecba11f55a34db29991d578c10ca8

- Put gist script into package
- Rename package to comply with r naming best practises 
- Split off large functions and partial apply to pass data from global scope 
- Remove hard coded file reading from main function and move to separate demo function

ConnorStroomberg committed:
Add raw script file , and escape dev.off

ConnorStroomberg committed:
Merge branch 'fix/description-may-only-have-one-maintainer' into spike/run-script

ConnorStroomberg committed:
Setup to demo script as is

- Rename variables to use more descriptive name.
- Comment out file handling part in script.
- Add run.demo function that asks for files and calls the script.

ConnorStroomberg committed:
Remove unused file handling logic

- File handling is now done outside of mozaiekberekentool.R.
- Add basic doc for main function
- regenarate man files

ConnorStroomberg committed:
Merge branch 'master' of github.com:molgenis/mosaicCalculator into spike/run-script

ConnorStroomberg committed:
Fix missing table

- Print result of MosaicCalculator function.

ConnorStroomberg committed:
Stip off main event processing functions into separate function, outside of scope of main function ( to prepapare for unit testing and improve redability off main algorithm).

- move even handeling funtions to events.R
- move outlier removal function to outliers.R

ConnorStroomberg committed:
Rename package to comply with R package nameing best practices

ConnorStroomberg committed:
remove unneeded dev.off

ConnorStroomberg committed:
Merge branch 'master' into spike/run-script

ConnorStroomberg committed:
Expose mosaicCalculator function using roxygen

- update namespace
- define Imports in Description file
- update testthat file with new package name

ConnorStroomberg committed:
Fix check errors and warnings

- add missing param docs
- add missing imports ( todo sort out more fine grained imports using importFrom)
- Fix errors in docs ( could not handle '%' in description)

ConnorStroomberg committed:
Remove fixed output file and replace with param

(needed for molgenis integration)

ConnorStroomberg committed:
Print all output to pdf

ConnorStroomberg committed:
Format result to nice pdf format

- Split of print function
- Preformat result before printing
- Add imports and docs

marikaris reviewed (commented):
ERROR: dependency ‘gridExtra’ is not available for package ‘mosaicr’

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+#'
+#Versionnumber: 0.6.0.0
+MosaicCalculator <- function(exp.nr, gender, snpm.data, deviations, outfile){
+  print("Dank u voor het kiezen van Mozaiek Bereken Tool, ontwikkeld door Robert Sietsma")
```
Please remove this

connoratrug replied:
done

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+  #Set the option of the output decimal to comma
+  options(OutDec = ",")
+  
+  cat("Ombouwen van deviations naar bruikbare parameters.","\n")
```
Can you translate this?

connoratrug replied:
done

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+MosaicCalculator <- function(exp.nr, gender, snpm.data, deviations, outfile){
+  print("Dank u voor het kiezen van Mozaiek Bereken Tool, ontwikkeld door Robert Sietsma")
+  #Set the option of the output decimal to comma
+  options(OutDec = ",")
```
I think for international purposes we want to use a . as separator for decimals?

connoratrug replied:
I don't know, i suggest we check with Marloes and Lennart, i suggest to leave is as is for now until we know what the users would like, i also do not know is the string processing in the script depends of the output decimal being an comma.

marikaris replied:
Agree

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+  cat("Ombouwen van deviations naar bruikbare parameters.","\n")
+  eventsoutput <- process.deviation.line(deviations, gender)
+
+  cat("Berekenen van kwaliteit aspecten array.","\n")
```
Can you translate this?

connoratrug replied:
done

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+  aantalevents <- nrow(eventsoutput)
+  line1 <- paste(aantalevents, " events die groter zijn dan 150kb, meer dan 10 probes hebben en LOH/AO gebieden groter dan 5Mb.", sep = "")
+  cat(line1, "\n")
+  cat("Bezig met het filteren en berekenen van de overgebleven events, een ogenblik geduld alstublieft...","\n")
```
Can you translate this?

connoratrug replied:
done

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+
+  #Returning the number of rows that are about to be used in the apply calculation
+  aantalevents <- nrow(eventsoutput)
+  line1 <- paste(aantalevents, " events die groter zijn dan 150kb, meer dan 10 probes hebben en LOH/AO gebieden groter dan 5Mb.", sep = "")
```
Can you translate this?

connoratrug replied:
done

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("SNPs buiten afwijkingen: ", SNPs_used_in_averageBAF))
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("Gemiddelde BAF: ", meanaverageBAF))
```
Can you translate this?

connoratrug replied:
done

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+  result.lines <- c(result.lines, paste(aantalevents, "events that a larger then 150kb."))
+  result.lines <- c(result.lines, "More then 10 probes have a LOH/AO area larger then 5Mb.")
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("SNPs buiten afwijkingen: ", SNPs_used_in_averageBAF))
```
Can you translate this?

connoratrug replied:
done

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("STDev BAF: ", sdaverageBAF))
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("Gemiddelde afwijking van BAF 0.5: ", MADaverageBAF))
```
Can you translate this?

connoratrug replied:
done

marikaris commented on R/mozaiekberekentool.R:
```diff
@@ -0,0 +1,158 @@
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("Gemiddelde afwijking van BAF 0.5: ", MADaverageBAF))
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("Correctiefactor: ", correctionfactor))
```
Can you translate this?

connoratrug replied:
done

marikaris reviewed (changes requested):
Please rename the "mozaiekberekentool.R" to something english

Please remove the "square" file

connoratrug commented:
> ERROR: dependency ‘gridExtra’ is not available for package ‘mosaicr’

Please install it, or use devtools install all (```devtools::install_deps(dependencies = TRUE)```)

ConnorStroomberg committed:
process review comments;

- Rename main function file to english filename
- Remoeve 'square' demo function
- Remove 'square' unit test example
- Update docs

ConnorStroomberg committed:
Process review comments;

- Remove welcome message

ConnorStroomberg committed:
Process review comments:

- Tranlate dutch output messages to englisch

ConnorStroomberg committed:
Process review comments:

- Translate Dutch output and dataframe labels into English.

marikaris commented on R/events.R:
```diff
@@ -0,0 +1,477 @@
+
+  #Homozygous copy losses disturb the BAF signal
+  if(CNVType == "Homozygous Copy Loss"){
+    return(c(OCR,"Homozygote deletion","NA","NA","NA","NA","NA","NA","NA","NA"))
```
"Homozygote" is still Dutch, should be "Homozygous"

connoratrug replied:
done

ConnorStroomberg committed:
fix tranlation
